File: Sort a linked list of 0s, 1s and 2s.cpp
class Solution {
  public:
    Node* segregate(Node* head) {
        Node* curr = head;
        int z = 0, o = 0, t = 0;
        while (curr != nullptr){
            if (curr -> data == 1) o += 1;
            else if (curr -> data == 2) t += 1;
            else z += 1;
            curr = curr -> next;
        }
        while (z--) cout << 0 << " ";
        while (o--) cout << 1 << " ";
        while (t--) cout << 2 << " ";
        return nullptr;
    }
};

File: Sort a linked list of 0s, 1s and 2s.js
class Solution {
  segregate(head) {
    let initAns = [];
    let news = head;
    while (news) {
      initAns.push(news.data);
      news = news.next;
    }

    const reducedValue = initAns.sort((a, b) => a - b);
    let current = head;
    let cou = 0;
    while (current) {
      current.data = reducedValue[cou];
      cou += 1;
      current = current.next;
    }

    return head;
  }
}
